ni/github.py

```diff
@@ -89,6 +91,12 @@ def __init__(self, server, event, request):
             raise abc.ResponseExit(
                     status=http.HTTPStatus.UNSUPPORTED_MEDIA_TYPE, text=msg)
 
+        binary_content = await request.read()
+
+        if not Host._verify_signature(server.contrib_secret_token(), binary_content,
```

Comment:
If `server.secret_token()` is `None`, then don't bother verifying.

The function call to `server` also doesn't match what is in the ABC.